components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/factory/RegistrationRequestFactory.java

```diff
@@ -117,6 +119,13 @@ public void create(IdentityRequest.IdentityRequestBuilder builder, HttpServletRe
             if (obj != null && obj instanceof JSONArray) {
                 JSONArray redirectUris = (JSONArray) obj;
                 for (int i = 0; i < redirectUris.size(); i++) {
+                    try{
+                        URL redirectURL = new URL(redirectUris.get(i).toString());
+                    }
+                    catch (MalformedURLException e){
+                        String errorMessage = "The redirection URI " + redirectUris.get(i).toString() + " is not allowed by this Server";
+                        throw IdentityException.error(FrameworkClientException.class, errorMessage, e);
+                    }
                     registrationRequestProfile.getRedirectUris().add(redirectUris.get(i).toString());
```

Comment:
Can improve the code by defining a variable for 'redirectUris.get(i).toString()'  and using same in both the places line 123 and 129.